smurfs/src/components/App.js

```diff
@@ -1,22 +1,76 @@
 import React, { Component } from 'react';
 import './App.css';
+import { connect } from "react-redux";
+// import { Button } from "reactstrap";
+
+
+import { getSmurfs, addSmurf } from "../actions";
 /*
  to wire this component up you're going to need a few things.
  I'll let you do this part on your own. 
  Just remember, `how do I `connect` my components to redux?`
  `How do I ensure that my component links the state to props?`
  */
 class App extends Component {
+
+  state = {
+    
+      name: '',
+      age: '', 
+      height: ''
+    
+  }
+
+  onClick = e => {
+    e.preventDefault();
+    const newSmurf = { name: this.state.name, 
+                        age: this.state.age, 
+                      height: this.state.height}
+    this.props.addSmurf(newSmurf)
+    this.setState({name: '', age: '', height: ''})
+  }
+
+  handleChange = e => {
+    const { name, value } = e.target;
+    this.setState({
+      [name]: value
+    });
+  };
   render() {
     return (
       <div className="App">
-        <h1>SMURFS! 2.0 W/ Redux</h1>
-        <div>Welcome to your Redux version of Smurfs!</div>
-        <div>Start inside of your `src/index.js` file!</div>
-        <div>Have fun!</div>
+       {this.props.smurfs.map(smurf => {
+         return(
+           <div>
+          <h1>{smurf.name}</h1>
+          <ul>
+            <li>Age: {smurf.age}</li>
+            <li>Height: {smurf.height}</li>
+          </ul>
+          </div>
+         );
+       })}
+      
+         <input onChange={this.handleChange} type="text" name="name" placeholder="Name" />
+         <input onChange={this.handleChange}  type="text" name="age" placeholder="Age" />
+         <input onChange={this.handleChange}  type="text" name="height" placeholder="Height" />
+         <button  onChange={this.handleChange}  onClick={this.onClick}>Add Smurf</button>
+         
+       
       </div>
     );
+  };
+
+
+componentDidMount(){
+  this.props.getSmurfs();
+}
+}
+const mapStateToProps = state => {
+  return{
+    smurfs: state.smurfs
   }
 }
-
-export default App;
+export default connect(
+  mapStateToProps,
+  { getSmurfs, addSmurf })(App);
```

Comment:
Everything looks good in this component as well, some indentation issues make it . bit harder to read and follow the flow of the information, I'd recommend the use of any extension for VSCODE that can help you with this, prettier is a very popular one, other than that everything is fine, well use of the connect function and mapStateToProps, you connected your component to the store and the ctions creators properly. Good job.